pkg/git/provider_gitlab.go

```diff
@@ -47,22 +51,64 @@ func NewGitlabProvider(baseURL string) (Provider, error) {
 	}, nil
 }
 
-func (g *gitlab) Type() ProviderType {
-	return g.providerType
-}
-
 func (g *gitlab) BaseURL() string {
 	urlClone := *g.apiURL
 	urlClone.Path = ""
 	urlClone.RawQuery = ""
 	return urlClone.String()
 }
 
-func (g *gitlab) VerifyToken(ctx context.Context, tokenType TokenType, token string) error {
-	log.G(ctx).Debug("Skip verifying token for gitlab, to be implemented later")
+func (g *gitlab) SupportsMarketplace() bool {
+	return false
+}
+
+func (g *gitlab) Type() ProviderType {
+	return g.providerType
+}
+
+func (g *gitlab) VerifyRuntimeToken(ctx context.Context, token string) error {
+	return g.checkApiScope(ctx, token)
+}
+
+func (g *gitlab) VerifyUserToken(ctx context.Context, token string) error {
+	return g.checkReadRepositoryScope(ctx, token)
+}
+
+func (g *gitlab) checkApiScope(ctx context.Context, token string) error {
+	res, err := g.request(ctx, token, http.MethodPost, "projects")
+	if err != nil {
+		return fmt.Errorf("failed checking api scope: %w", err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusBadRequest {
+		return errors.New("git-token is invalid or missing required \"api\" scope")
+	}
+
 	return nil
 }
 
-func (g *gitlab) SupportsMarketplace() bool {
-	return false
+func (g *gitlab) checkReadRepositoryScope(ctx context.Context, token string) error {
+	res, err := g.request(ctx, token, http.MethodHead, "projects")
```

Comment:
HEAD to projects.
if it returns 200 - the token has "repo_read" scope
otherwise - the token does not have the scope